ldap-tls/auth.go

```diff
@@ -3,76 +3,114 @@ package main
 import (
 	"crypto/tls"
 	"crypto/x509"
-	"errors"
 	"fmt"
 	"net/http"
 
-	"maverics/app"
-	"maverics/ldap"
-	"maverics/log"
-	"maverics/secret"
-	"maverics/session"
-)
-
-const (
-	idpName          = "ldap"
-	ldapURL          = "ldap://ldap.examples.com"
-	ldapServerName   = "ldap.examples.com"
-	ldapBaseDN       = "ou=People,dc=examples,dc=com"
-	ldapCASecretName = "ldapCA"
+	ldap3 "github.com/go-ldap/ldap/v3"
+	"github.com/strata-io/service-extension/orchestrator"
 )
 
 // IsAuthenticated determines if the user has been authenticated.
-func IsAuthenticated(ag *app.AppGateway, rw http.ResponseWriter, req *http.Request) bool {
-	log.Debug("se", "determining if user is authenticated")
+func IsAuthenticated(api orchestrator.Orchestrator, _ http.ResponseWriter, _ *http.Request) bool {
+	logger := api.Logger()
+	session, err := api.Session()
+	if err != nil {
+		logger.Error("se", "unable to retrieve session", "error", err.Error())
+		return false
+	}
+	metadata := api.Metadata()
+	idpName := metadata["idpName"].(string)
 
-	if session.GetString(req, fmt.Sprintf("%s.authenticated", idpName)) == "true" {
+	logger.Debug("se", "determining if user is authenticated")
+
+	isAuthenticated, err := session.GetString(fmt.Sprintf("%s.authenticated", idpName))
+	if err != nil {
+		logger.Error(
+			"se", fmt.Sprintf("unable to retrieve session value '%s.authenticated'", idpName),
+			"error", err.Error(),
+		)
+		return false
+	}
+	if isAuthenticated == "true" {
 		return true
 	}
 
 	return false
 }
 
 // Authenticate authenticates the user against the IDP.
-func Authenticate(ag *app.AppGateway, rw http.ResponseWriter, req *http.Request) error {
+func Authenticate(api orchestrator.Orchestrator, rw http.ResponseWriter, req *http.Request) {
+	logger := api.Logger()
+	session, err := api.Session()
+	if err != nil {
+		logger.Error("se", "unable to retrieve session", "error", err.Error())
+		http.Error(rw, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+	secretP, err := api.SecretProvider()
```

Comment:
Nit, using `secretProvider` as the variable may be a bit more clear:
```suggestion
	secretProvider, err := api.SecretProvider()
```